File: Assets/scripts/bossBehavor.js
#pragma strict
var bossLife:int = 30;
private var  timeTemp:float=0;
var bosszidan:GameObject;
function Start () {

}

function Update () {
	timeTemp+=Time.deltaTime;
}

function OnCollisionEnter2D(col:Collision2D){
if(col.gameObject.name=="zidan_t(Clone)"){
	bossLife = bossLife-1;
}
if(bossLife<0){
	gameObject.Find("wuPin_Builder").GetComponent(AddLife).isHeroLifeRun=true;
 	Destroy(this.gameObject);
	
}
if(timeTemp>0.5){
	var n:GameObject = Instantiate(bosszidan,Vector3(transform.position.x,transform.position.y-1.5,0),transform.rotation);
	timeTemp=0;
}

	

}

File: Assets/scripts/Cs/BossBehavor.cs
using UnityEngine;
using System.Collections;

public class BossBehavor : MonoBehaviour {
    public int _bossLife = 30;
    public GameObject bosszidan;


    private float _timeTemp = 0;
    
	// Use this for initialization
	void Start () {
	
	}
	
	// Update is called once per frame
	void Update () {
        _timeTemp += Time.deltaTime;
	}
    void OnCollisionEnter2D(Collision2D col) { 
        if(col.gameObject.name=="zidan(Clone)"){
            _bossLife = _bossLife - 1;
        }
        if(_bossLife<0){
            GameObject.Find("wuPin_Builder").GetComponent<AddLife>().isHeroLifeRun = true;
            Destroy(this.gameObject);
        }
        if(_timeTemp>0.5){
            GameObject n = Instantiate(bosszidan,new Vector3(transform.position.x,transform.position.y-1.5f,0),transform.rotation)as GameObject;
        }


    }

}
